lib/vrl/stdlib/src/keys.rs

```diff
@@ -0,0 +1,71 @@
+use ::value::Value;
+use vrl::function::ArgumentList;
+use vrl::function::Compiled;
+use vrl::function::Example;
+use vrl::function::FunctionCompileContext;
+use vrl::prelude::*;
+use vrl::state::TypeState;
+use vrl::Expression;
+use vrl::Function;
+
+fn keys(value: Value) -> Resolved {
+    let object = value.try_object()?;
+    let keys = object.into_keys().map(Value::from);
+    Ok(Value::Array(keys.collect()))
+}
+
+#[derive(Debug)]
+pub struct Keys;
+
+impl Function for Keys {
+    fn identifier(&self) -> &'static str {
+        "keys"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::OBJECT,
+            required: true,
+        }]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "get keys",
+                source: r#"keys({"key1": "val1", "key2": "val2"})"#,
+                result: Ok(r#"["key1", "key2"]"#),
+            },
+            Example {
```

Comment:
Maybe change the structure of the 2nd example a bit. Right now it's identical to the 1st, just with different field names. A nested object could be useful to show only the "top-level" keys are returned.

Reply:
Same comment for the `values` function.